Nodejs/Product/Nodejs/Commands/DiagnosticsCommand.cs

```diff
@@ -141,9 +143,73 @@ internal sealed class DiagnosticsCommand : Command {
                     jsAnalyzer.DumpLog(writer);
                 }
             }
+            res.AppendLine("Files:");
+            res.AppendLine(Indent(4, GetProjectFileInfo(project)));
+            return res.ToString();
+        }
+
+        /// <summary>
+        /// Stores information about a collection of files of a given type.
+        /// </summary>
+        private class FileTypeInfo {
+            private int _count = 0;
+            private int _maxLineLength = 0;
+            private int _totalLineLength = 0;
+
+            public int Count {
+                get { return _count; }
+            }
+
+            public int MaxLineLength {
+                get { return _maxLineLength; }
+            }
+
+            public int AverageLineLength {
+                get { return _count > 0 ? _totalLineLength / _count : 0; }
+            }
+
+            public void UpdateForFile(string file) {
+                try {
+                    int length = File.ReadLines(file).Count();
+                    ++_count;
+                    _totalLineLength += length;
+                    _maxLineLength = Math.Max(_maxLineLength, length);
+                } catch (IOException) {
+                    // noop
+                }
+            }
+        }
+
+        private static string GetProjectFileInfo(Project.NodejsProjectNode project) {
+            var fileTypeInfo = new Dictionary<string, FileTypeInfo>();
+            foreach (var node in project.DiskNodes) {
+                if (node.Value.ItemNode?.IsExcluded ?? true) {
+                    continue;
+                }
+                var file = node.Key;
+                var ext = Path.GetExtension(file).ToLowerInvariant();
```

Comment:
We should abort here if the extension doesn't match a particular file type otherwise we might end up reading enormous binary files or something. We mostly care about .js, .jsx, .tsx, .ts, .d.ts, .html